### ISSUE TITLE
Bug on multiple fields dependsOn for one field

### ISSUE THREAD

**Issue by rdemaret** [laravel/nova-issues#3870]
- Laravel Version: 9.2
- Nova Version: 4.0.6
- PHP Version: 8.1.3
- Database Driver & Version: MySql

Hello,
I've a problem with dependsOn with nova. I've a boolean field, when it's true it must set readonly one field and make the other not. When false, it is the opposite. But the second Field label get the first one label and don't change readonly. It seems to be a bug with dependsOn.

```php
            Boolean::make('URL ?', 'is_link')->sortable(),

            File::make('Fichier', 'path')
                ->disk('public')
                ->path('documents/' . uniqid())
                ->storeAs(function (Request $request) {
                    return $request->name . '.' . $request->path->extension();
                })
                ->readonly(false)
                ->dependsOn(['is_link'],
                    function (File $field, NovaRequest $request, FormData $formData) {
                    if ($formData->is_link){
                        $field->readonly();
                    }
                }),

            URL::make('URL', 'path')
                ->readonly()
                ->dependsOn(['is_link'],
                    function (URL $field, NovaRequest $request, FormData $formData) {
                        if ($formData->is_link){
                            $field->readonly(false);
                        }
                    }),`
```

**Comment by crynobone** [comment]
Duplicate of #3863

PR has been merged pending next release.

**Comment by rdemaret** [comment]
Still not fixed with 4.1 ?

**Comment by elizdlt202** [comment]
still not fixed version 4.2.3 ?

**Comment by crynobone** [comment]
Submit new issue with complete application example. If the issue persist clearly I'm missing a critical use case that can only be available with complete app. See https://github.com/nova-issues for reference.
